src/main/java/randoop/main/GenInputsAbstract.java

```diff
@@ -1148,6 +1170,117 @@ private static boolean shouldOmitClass(String classname) {
     }
   }
 
+  /**
+   * Returns classes in the given package. Does not include classes in sub-packages. Ignores
+   * non-accessible classes.
+   *
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes in package {@code packageName}
+   */
+  private static Set<@ClassGetName String> getClassnamesFromPackage(
+      String packageName, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> classnames = new HashSet<>();
+
+    for (String path : Globals.getClassPath().split(File.pathSeparator)) {
+      File location = new File(path);
+      if (location.isFile() && location.getName().endsWith(".jar")) {
+        classnames.addAll(getClassesWithPackageFromJar(location, packageName, accessibility));
+      } else if (location.isDirectory()) {
+        classnames.addAll(getClassesWithPackageFromDirectory(location, packageName, accessibility));
+      }
+    }
+
+    return classnames;
+  }
+
+  /**
+   * Searches in the given directory for classes with the given package. Directory must be root
+   * directory containing class files, e.g. {@code build/classes/} for gradle projects or {@code
+   * target/classes/} for maven projects.
+   *
+   * @param directory a directory
+   * @param packageName a package name
+   * @param accessibility the accessibility predicate
+   * @return classes with the given package that were found in the specified directory
+   */
+  private static Set<@ClassGetName String> getClassesWithPackageFromDirectory(
+      File directory, String packageName, AccessibilityPredicate accessibility) {
+    String packageNameAsFile = packageName.replace(".", File.separator);
+    File packageDirectory =
+        new File(directory.getPath().concat(File.separator).concat(packageNameAsFile));
+    if (packageDirectory.exists() && packageDirectory.isDirectory()) {
+      Set<@ClassGetName String> classnames = new HashSet<>();
+      for (File file :
+          packageDirectory.listFiles(f -> f.isFile() && f.getName().endsWith(".class"))) {
+
+        String relativePath =
+            file.getPath()
+                .substring(directory.getAbsolutePath().length() + File.separator.length());
```

Comment:
I believe that if `directory` is a relative pathname, then `file` will be too, in which case trimming off enough characters for `directory`'s absolute pathname would be incorrect.  Could you double-check that?
I wonder if the [`relativize`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/nio/file/Path.html#relativize(java.nio.file.Path)) method would be useful here instead of doing string manipulation.  String manipulation is a code smell.